### PR TITLE
Fix a few uses_dynamic_as_bottom errors around FormatFunction

### PR DESCRIPTION
**Pull request by srawlins**
This PR basically forces every function assigned as a `FormatFunction` to understand that it needs to accept `dynamic` arguments, as the signature of `FormatFunction` is:

```dart
typedef String FormatFunction(value);
```

The alternative to this PR is something far hairier. Specifically, we have the following declarations:

```dart
abstract class Scale {
  FormatFunction createTickFormatter([String format]);
}

class LinearScale implements Scale {
  FormatFunction createTickFormatter([String formatStr]);
}

class TimeScale extends LinearScale {
  // A function of type '(DateTime) → String' can't be assigned to a variable of type '(dynamic) → String'
  FormatFunction createTickFormatter([String format]);
}
```

If we instead want to type annotate FormatFunction as:

```dart
typedef String FormatFunction<T>(T value);
```

Then we'd be able to type annotate `createTickFormatter`, but not in a way that `TimeScale` could still extend `LinearScale`. This would be a much bigger change that I should not author. I suggest we go with this PR, at least for now.

### CODE REVIEW

File: lib/core/scales/ordinal_scale.dart [hunk 1]
```diff
@@ -95,7 +95,7 @@ class _OrdinalScale implements OrdinalScale {
 
   @override
   FormatFunction createTickFormatter([String format]) =>
-      (String s) => identityFunction<String>(s);
+      (String s) => identityFunction<String>(s as String);
```
Comment (leafpetersen) [inline lib/core/scales/ordinal_scale.dart]: should be `(dynamic s) => `?

Comment (srawlins) [reply lib/core/scales/ordinal_scale.dart]: Done.

**Comment by psunkari** [comment]
Thank you Sam for fixing these.
I understand how difficult it can be to go through (and fix) a code base that someone else authored.

**Comment by srawlins** [comment]
@psunkari Thanks for reviewing! The real difficulty is touching a code base without tests :stuck_out_tongue_winking_eye: Internal CI gets the job done though.

**Pull request merged**